lib/csv_to_popolo.rb

```diff
@@ -339,30 +353,72 @@ def given?(key)
       @r.key?(key) && !@r[key].nil? && !@r[key].empty?
     end
 
+    def split_cell(key)
+      separator = MODEL[key][:multivalue_separator]
+      if separator
+        @r[key].split(separator)
+      else
+        [@r[key]]
+      end
+    end
+
+    def set_cell_from_values(key, values)
+      separator = MODEL[key][:multivalue_separator]
+      if separator
+        @r[key] = values.join(separator)
+      else
+        if values.length == 0
+          @r[key] = ''
+        elsif values.length > 1
+          raise "No multivalue_separator for #{key} but setting a cell with multiple values"
+        else
+          @r[key] = values[0]
+        end
+      end
+    end
+
+    def keys_with_values_for_type(type)
+      MODEL.select { |_, v| v[:type] == type }
+           .map    { |k, _| k }
+           .select { |key| given? key }
+    end
+
     def contact_details
       # Standardise Twitter handles
       if given? :twitter
-        @r[:twitter] = TwitterUsernameExtractor.extract(@r[:twitter]) rescue nil
+        twitter_usernames = split_cell(:twitter).map {
+          |t| TwitterUsernameExtractor.extract(t) rescue nil
+        }
+        set_cell_from_values(:twitter, twitter_usernames)
       end
 
-      contacts = MODEL.select { |_, v| v[:type] == 'contact' }
-                 .map    { |k, _| k }
-                 .select { |type| given? type }
-                 .map    { |type| { type: type.to_s, value: @r[type] } }
-                 .compact
+      contacts = []
+      keys_with_values_for_type('contact').each do |key|
+        split_cell(key).each do |value|
+          contacts.push({ type: key.to_s, value: value })
```

Comment:
Ah, I tried removing the bananas which gives a syntax error, when I should have been removing the moustaches :)